spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplateIntegrationTests.java

```diff
@@ -1355,9 +1366,15 @@ public Message handleMessage(Message message) {
 			Object result = this.template.convertSendAndReceive("foo");
 			container.stop();
 			assertEquals("FOO", result);
-			assertThat(replyToWas.get(), startsWith(Address.AMQ_RABBITMQ_REPLY_TO));
+			if (tempQueue) {
```

Comment:
Understood, I see that big part of code should be changed in that case. I just prefer BDD approach(given, when, then), it looks much cleaner.